Avalonia.sln

```diff
@@ -12,6 +12,11 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Avalonia.Direct2D1", "src\Windows\Avalonia.Direct2D1\Avalonia.Direct2D1.csproj", "{3E908F67-5543-4879-A1DC-08EACE79B3CD}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Avalonia.Controls", "src\Avalonia.Controls\Avalonia.Controls.csproj", "{D2221C82-4A25-4583-9B43-D791E3F6820C}"
+	ProjectSection(ProjectDependencies) = postProject
```

Comment:
I think it should be built before Avalonia.Base.

It's weird how .NET 7 SDK somehow skips explicit ProjectReference nodes

Reply:
@kekekeks  do you mind to expands ont the ProjectReference node ? when looking at that code change, I was wondering why you didn't use project reference but use solution project dependencies as by experience we do not use this on our side as tools which only look at projects will not be able to use this

Reply:
We _are_ using ProjectReference nodes and it used to work perfectly fine with .NET 6 SDK.

https://github.com/AvaloniaUI/Avalonia/blob/master/build/SourceGenerators.props#L3
https://github.com/AvaloniaUI/Avalonia/blob/master/build/DevAnalyzers.props

Reply:
Not really, I encountered a problem several times that I had to explicitly rebuild e.g. Avalonia.Build.Tasks after switching branch before we move to .net 7.